compiler-core/src/type_/tests/warnings.rs

```diff
@@ -1138,6 +1138,21 @@ fn unused_alias_warning_test() {
     );
 }
 
+#[test]
+fn used_type_with_import_alias_no_warning_test() {
+    assert_warnings_with_imports!(
+        ("gleam/foo", "pub type A");
+        r#"
+            import gleam/foo as bar
+            fn fun(a: bar.A) { a }
```

Comment:
Could you make this function public to remove the other warning please? It's not very clear to me what this is testing currently.

Reply:
>  I had to use assert_warnings_with_imports for the test because assert_no_warnings! was not failing without the fix!

This is the issue, I had to do this so the test would fail without the fix!

Reply:
oh! How come it wasn't failing? 🤔

Reply:
The weird thing is, even if the module name in the first argument and the one imported in the second argument don't match, it still does not complain. It only started complaining when there was syntax errors I think, but it seems like there are a few issues that `assert_no_warnings!` simply ignores!

Reply:
We will need to use `assert_no_warnings` as this test is unclear.